logstash-core/lib/logstash/agent.rb

```diff
@@ -262,4 +268,8 @@ def clean_state?
   def reset_collector
     LogStash::Instrument::Collector.instance.clear
   end
+
+  def setting(key)
```

Comment:
imo it's just a matter of need depending on the class.
Usually the pattern would be `LogStash::SETTINGS.get(key)`, but to ease testing and avoid this direct access we pass the Settings instance as an argument, so that turns into `@settings.get(key)`, but in classes that make a very high use of settings like the runner I wanted to be less verbose, hence the accessor method, to reduce to `setting(key)`.

From this perspective, what do you suggest?